vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -1012,6 +1024,12 @@ func resourceVSphereVirtualMachineCustomizeDiff(_ context.Context, d *schema.Res
 		return err
 	}
 
+        // Check the power state diff for VM
```

Comment:
States names should reflect the following, _which are also seen in PowerCLI:_

- `PoweredOn`
- `PoweredOff`
- `Suspended`

Reply:
Resolved in https://github.com/hashicorp/terraform-provider-vsphere/pull/1407/commits/4925157671e97d52889181d977cad8123b92473d.